Source/WebCore/css/CSSProperties.json

```diff
@@ -1240,28 +1347,50 @@
             }
         },
         "text-spacing-trim": {
+            "animation-type": "discrete",
             "inherited": true,
+            "initial": "space-all",
+            "initial-comment": "Current spec initial value is 'normal'",
             "values": [
-                "auto",
                 "space-all",
-                "trim-all"
+                {
+                    "value": "normal",
+                    "status": "unimplemented"
+                },
+                {
+                    "value": "space-first",
+                    "status": "unimplemented"
+                },
+                {
+                    "value": "trim-start",
+                    "status": "unimplemented"
+                },
+                {
+                    "value": "trim-both",
+                    "status": "unimplemented"
+                },
+                "trim-all",
+                "auto"
             ],
             "codegen-properties": {
                 "settings-flag": "cssTextSpacingTrimEnabled",
                 "font-property": true,
                 "high-priority": true,
                 "sink-priority": true,
-                "converter": "TextSpacingTrim",
-                "parser-grammar": "space-all | trim-all | auto"
+                "style-builder-converter": "TextSpacingTrim",
+                "parser-grammar": "<<values>>",
+                "parser-grammar-comment": "Current spec grammar is 'space-all | normal | space-first | trim-start | trim-both | trim-all | auto'"
             },
             "status": "in development",
             "specification": {
                 "category": "css-text",
-                "non-canonical-url": "https://www.w3.org/TR/css-text-4/#text-spacing-property"
+                "url": "https://drafts.csswg.org/css-text-4/#propdef-text-spacing-trim"
             }
         },
         "text-autospace": {
+            "animation-type": "discrete",
             "inherited": true,
+            "initial": "normal",
```

Comment:
This doesn't match `initialTextAutospace` (see https://github.com/WebKit/WebKit/pull/40309 ).

Reply:
Fixed. Thanks.